include/alpaka/core/ApiCudaRt.hpp

```diff
@@ -253,6 +254,17 @@ namespace alpaka
             return ::cudaHostUnregister(ptr);
         }
 
+        static inline Error_t launchCooperativeKernel(
+            void const* func,
+            dim3 gridDim,
+            dim3 blockDim,
+            void** args,
+            size_t sharedMem,
+            Stream_t stream)
+        {
+            return ::cudaLaunchCooperativeKernel(func, gridDim, blockDim, args, sharedMem, stream);
+        }
+
```

Comment:
Could you change this to be templated on the `func` argument ?
```suggestion
        template <typename TFunc>
        static inline Error_t launchCooperativeKernel(
            TFunc func,
            dim3 gridDim,
            dim3 blockDim,
            void** args,
            size_t sharedMem,
            Stream_t stream)
        {
            return ::cudaLaunchCooperativeKernel(func, gridDim, blockDim, args, sharedMem, stream);
        }

```

Same for the HIP implementation.

Reply:
Does it really matter, if the underlying function wants void* anyway?